packages/cli/openapi-parser/src/v3/converters/example/ExampleTypeFactory.ts

```diff
@@ -372,23 +380,51 @@ export class ExampleTypeFactory {
         }
     }
 
-    private hasExample(schema: SchemaWithExample): boolean {
+    private hasExample({
+        schema,
+        options,
+        depth
+    }: {
+        schema: SchemaWithExample;
+        options: ExampleTypeFactory.Options;
+        depth: number;
+    }): boolean {
+        if (this.exceedsMaxDepth(depth, options)) {
+            return false;
+        }
```

Comment:
the idea here was that if an example is present in the OAI spec, we should ignore max depth